src/instructlab/model/serve.py

```diff
@@ -191,3 +194,13 @@ def serve(
     finally:
         backend_instance.shutdown()
         raise click.exceptions.Exit(0)
+
+
+def warn_for_unsuported_backend_param(
+    ctx: click.Context,
+):
+    for param in ["gpu_layers", "num_threads", "max_ctx_size"]:
+        if ctx.get_parameter_source(param) == click.core.ParameterSource.COMMANDLINE:
```

Comment:
why would you only log when it's used via command line?  is it not possible to set in the config file?

I know click supports setting flags via env vars too, but I don't think we have that turned on ...

Reply:
It's simpler change to first test the CLI and I [mentioned](https://github.com/instructlab/instructlab/pull/1775#issuecomment-2235962348) that checking the config could be done as well. :)

Reply:
You could just check if any values are different than the defaults and check all cases at once?

Reply:
> You could just check if any values are different than the defaults and check all cases at once?

But that wouldn't solve the case where someone passes a flag pointing to the same value as the default one :/

Reply:
I guess so. It seems like a pretty small change to add in any case.  I'm not blocking the PR on it, though.

Reply:
Actually all gpu_layers, max_ctx_size are part of the llama_cpp section, so it's clear that they are not part of vllm.